mysql-test/suite/period/r/delete.result

```diff
@@ -0,0 +1,273 @@
+select *, unix_timestamp(row_start),
+unix_timestamp(row_end) from t for system_time all
+order by row_end, s, e;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	2018-12-12	3141592.000000	314159265.358979
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+2018-01-01	2018-12-12	314159265.358979	2147483647.999999
+select * from log_tbl;
+log
+>DEL: 1999-01-01, 2018-12-12
+<DEL: 1999-01-01, 2018-12-12
+>INS: 1999-01-01, 2000-01-01
+<INS: 1999-01-01, 2000-01-01
+>INS: 2018-01-01, 2018-12-12
+<INS: 2018-01-01, 2018-12-12
+# system versioning with transactions
+create or replace table t (s date, e date,
+rs bigint(20) unsigned as row start invisible,
+re bigint(20) unsigned as row end invisible,
+period for apptime(s, e),
+period for system_time(rs, re)) with system versioning engine=InnoDB;
+insert into t values('1999-01-01', '2018-12-12'),
+('1999-01-01', '1999-12-12');
+select rs from t limit 1 into @ins_trx;
+Warnings:
+Warning	1287	'<select expression> INTO <destination>;' is deprecated and will be removed in a future release. Please use 'SELECT <select list> INTO <destination> FROM...' instead
+select re from t limit 1 into @trx_max;
+Warnings:
+Warning	1287	'<select expression> INTO <destination>;' is deprecated and will be removed in a future release. Please use 'SELECT <select list> INTO <destination> FROM...' instead
+select * from t;
+s	e
+1999-01-01	2018-12-12
+1999-01-01	1999-12-12
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select *, if(rs = @ins_trx, "OLD", "NEW"),
+if(re = @trx_max, "CURRENT", "HISTORY") from t for system_time all;
+s	e	if(rs = @ins_trx, "OLD", "NEW")	if(re = @trx_max, "CURRENT", "HISTORY")
+1999-01-01	2000-01-01	NEW	CURRENT
+1999-01-01	1999-12-12	OLD	CURRENT
+1999-01-01	2018-12-12	OLD	HISTORY
+2018-01-01	2018-12-12	NEW	CURRENT
+# same for trigger case
+delete from t;
+delete history from t;
+insert into t values('1999-01-01', '2018-12-12'),
+('1999-01-01', '1999-12-12');
+select rs from t limit 1 into @ins_trx;
+Warnings:
+Warning	1287	'<select expression> INTO <destination>;' is deprecated and will be removed in a future release. Please use 'SELECT <select list> INTO <destination> FROM...' instead
+select * from t;
+s	e
+1999-01-01	2018-12-12
+1999-01-01	1999-12-12
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select *, if(rs = @ins_trx, "OLD", "NEW"),
+if(re = @trx_max, "CURRENT", "HISTORY") from t for system_time all;
+s	e	if(rs = @ins_trx, "OLD", "NEW")	if(re = @trx_max, "CURRENT", "HISTORY")
+1999-01-01	2018-12-12	OLD	HISTORY
+1999-01-01	1999-12-12	OLD	CURRENT
+1999-01-01	2000-01-01	NEW	CURRENT
+2018-01-01	2018-12-12	NEW	CURRENT
+select * from log_tbl;
+log
+>DEL: 1999-01-01, 2018-12-12
+<DEL: 1999-01-01, 2018-12-12
+>INS: 1999-01-01, 2000-01-01
+<INS: 1999-01-01, 2000-01-01
+>INS: 2018-01-01, 2018-12-12
+<INS: 2018-01-01, 2018-12-12
+create or replace database test;
```

Comment:
## Bug 11: unexpected warning on correct argument
### Reproduce
```sql
create or replace table t (id int, s date, e date, period for apptime(s,e));
delete from t for portion of apptime from 777222333 to '2018-01-01';
show warnings;
```
### Result
```
+---------+------+------------------------------------------------+
| Level   | Code | Message                                        |
+---------+------+------------------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: '2018-01-01' |
+---------+------+------------------------------------------------+
```
### Expected
It should print warning about first argument, not the second as it works ok. In fact it should fail with error, because when it treats wrong type as a zero date it leads to unexpected result.

Reply:
Thanks again
Looks like this bug deserves a separate task. I've created [MDEV-17209](https://jira.mariadb.org/browse/MDEV-17209) for it.

BTW isn't it connected with #379?

Reply:
No, I don't think so. But it is reproducible in SV.